packages/backend-common/src/reading/BitbucketUrlReader.ts

```diff
@@ -150,4 +196,51 @@ export class BitbucketUrlReader implements UrlReader {
     }
     return `bitbucket{host=${host},authed=${authed}}`;
   }
+
+  private async getLastCommitShortHash(
+    gitUrl: parseGitUri.GitUrl,
+  ): Promise<String> {
+    const { name: repoName, owner, ref, protocol, resource } = gitUrl;
+
+    const isHosted = resource === 'bitbucket.org';
+
+    const branch = ref ? ref : 'master';
+    const commitsApiUrl = isHosted
+      ? `${protocol}://${this.config.apiBaseUrl}/repositories/${owner}/${repoName}/commits/${branch}`
+      : `${protocol}://${this.config.apiBaseUrl}/projects/${owner}/repos/${repoName}/commits/?until=${branch}`;
```

Comment:
As for the Bitbucket Server API, here are resources I found:
- https://developer.atlassian.com/server/bitbucket/reference/rest-api/;
- https://developer.atlassian.com/server/bitbucket/how-tos/command-line-rest/.